fastlane_bot/modes/triangle_multi.py

```diff
@@ -22,65 +20,141 @@ class ArbitrageFinderTriangleMulti(ArbitrageFinderTriangleBase):
-                continue
-            profit_src = -r.result
+    def assert_supported(self):
+        assert self.base_exchange == "carbon_v1", "base_exchange for `multi_triangle` mode must be `carbon_v1`"
 
-            # Get the cids
-            cids = [ti["cid"] for ti in trade_instructions_dic]
-
-            # Calculate the profit
-            profit = self.calculate_profit(src_token, profit_src, self.CCm, cids)
-            if str(profit) == "nan":
-                self.ConfigObj.logger.debug("profit is nan, skipping")
-                continue
+    def get_combos(self, flashloan_tokens: List[str], CCm: Any, arb_mode: str) -> Tuple[List[str], List[Any]]:
+        """
+        Get combos for triangular arbitrage
+
+        Parameters
+        ----------
+        flashloan_tokens : list
+            List of flashloan tokens
+        CCm : object
+            CCm object
+        arb_mode : str
+            Arbitrage mode
+
+        Returns
+        -------
+        combos : list
+            List of combos
 
-            # Handle candidates based on conditions
-            candidates += self.handle_candidates(
-                best_profit,
-                profit,
-                trade_instructions_df,
-                trade_instructions_dic,
-                src_token,
-                trade_instructions,
-            )
+        """
+        combos = []
+        if arb_mode in ["b3_two_hop"]:
+            combos = [
+                (tkn0, tkn1)
+                for tkn0, tkn1 in product(flashloan_tokens, flashloan_tokens)
+                # note that the pair is tkn0/tkn1, ie tkn1 is the quote token
+                if tkn0 != tkn1
+            ]
+        else:
+            all_base_exchange_curves = CCm.byparams(exchange=self.base_exchange).curves
+            for flt in flashloan_tokens:  # may wish to run this for one flt at a time
+                non_flt_base_exchange_curves = [
+                    x for x in all_base_exchange_curves if flt not in x.pair
+                ]
+                for non_flt_base_exchange_curve in non_flt_base_exchange_curves:
+                    target_tkny = non_flt_base_exchange_curve.tkny
+                    target_tknx = non_flt_base_exchange_curve.tknx
+                    base_exchange_curves = (
+                        CCm.bypairs(f"{target_tknx}/{target_tkny}")
+                        .byparams(exchange=self.base_exchange)
+                        .curves
+                    )
+                    if len(base_exchange_curves) == 0:
+                        continue
+
+                    base_direction_pair = base_exchange_curves[0].pair
+                    base_direction_one = [curve for curve in base_exchange_curves if curve.pair == base_direction_pair]
+                    base_direction_two = [curve for curve in base_exchange_curves if curve.pair != base_direction_pair]
+                    assert len(base_exchange_curves) == len(base_direction_one) + len(base_direction_two)
```

Comment:
cant really fail, no?

Reply:
Outside the scope of this PR (same code used prior to it), but good point regardless, so will remove